src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -4291,11 +4296,11 @@ void QgsPostgresSharedData::setFeaturesCounted( long count )
 }
 
 
-QgsFeatureId QgsPostgresSharedData::lookupFid( const QVariant &v )
+QgsFeatureId QgsPostgresSharedData::lookupFid( const QVariantList& v )
 {
   QMutexLocker locker( &mMutex );
 
-  QMap<QVariant, QgsFeatureId>::const_iterator it = mKeyToFid.constFind( v );
+  QMap<QVariantList, QgsFeatureId>::const_iterator it = mKeyToFid.constFind( v );
```

Comment:
QHash would be a better choice, I was just keen to understand the problem and fix it with the least amount of effort / changes ;-) QHash will just require our own qHash implementation for QVariant I suppose.